src/ExampleApp.cpp

```diff
@@ -117,143 +132,316 @@ ExampleApp::ExampleApp(Eegeo::EegeoWorld* pWorld,
+
+    glViewport(0, 0, m_screenPropertiesProvider.GetScreenProperties().GetScreenWidth()/2.0f, m_screenPropertiesProvider.GetScreenProperties().GetScreenHeight());
+
+    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentLeftCameraState(headTansform));
+
+    DrawEyeFromCameraState(dt, cameraState, eegeoWorld);
+
+}
+
+void ExampleApp::DrawRightEye (float dt, float headTansform[], Eegeo::EegeoWorld& eegeoWorld){
+    m_pExampleController->PreWorldDraw();
+
+    glViewport(m_screenPropertiesProvider.GetScreenProperties().GetScreenWidth()/2.0f, 0, m_screenPropertiesProvider.GetScreenProperties().GetScreenWidth()/2.0f, m_screenPropertiesProvider.GetScreenProperties().GetScreenHeight());
+
+    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentRightCameraState(headTansform));
+
+    DrawEyeFromCameraState(dt, cameraState, eegeoWorld);
+
+}
+
+void ExampleApp::DrawEyeFromCameraState(float dt, const Eegeo::Camera::CameraState& cameraState, Eegeo::EegeoWorld& eegeoWorld)
+{
     Eegeo::EegeoDrawParameters drawParameters(cameraState.LocationEcef(),
-                                              cameraState.InterestPointEcef(),
-                                              cameraState.ViewMatrix(),
-                                              cameraState.ProjectionMatrix(),
-                                              m_screenPropertiesProvider.GetScreenProperties());
-    
+    cameraState.InterestPointEcef(),
+    cameraState.ViewMatrix(),
+    cameraState.ProjectionMatrix(),
+    m_screenPropertiesProvider.GetScreenProperties());
+
+    Eegeo::v3 forward(m_pExampleController->GetOrientation().GetRow(2));
+    Eegeo::dv3 position(cameraState.LocationEcef() + (forward*50));
+
     eegeoWorld.Draw(drawParameters);
-    
+
     m_pExampleController->Draw();
-    
-    if (m_pLoadingScreen != NULL)
+}
+
+
+void ExampleApp::UpdateCardboardProfile(float cardboardProfile[])
+{
+	#ifdef CARDBOARD
+    m_pExampleController->UpdateCardboardProfile(cardboardProfile);
+    m_pVRDistortion->UpdateCardboardProfile(cardboardProfile);
+	#endif
+}
+
+void ExampleApp::MagnetTriggered()
+{
+
+}
+
+void ExampleApp::AppUpdate(float dt, const Eegeo::Camera::CameraState& cameraState, Eegeo::EegeoWorld& eegeoWorld)
+{
+	#ifdef CARDBOARD
+    if(m_pVRSkybox->IsShowing())
     {
-        m_pLoadingScreen->Draw();
+        m_pVRSkybox->Stop();
     }
+    m_vrModeTracker.ExitVRMode();
+	#endif
+    Eegeo::Streaming::IStreamingVolume& streamingVolume(m_pExampleController->GetCurrentStreamingVolume());
+
+    Eegeo::EegeoUpdateParameters updateParameters(dt,
+    cameraState.LocationEcef(),
+    cameraState.InterestPointEcef(),
+    cameraState.ViewMatrix(),
+    cameraState.ProjectionMatrix(),
+    streamingVolume,
+    m_screenPropertiesProvider.GetScreenProperties());
+    eegeoWorld.Update(updateParameters);
+}
+
+void ExampleApp::VRUpdate(float dt, const Eegeo::Camera::CameraState& cameraState, Eegeo::EegeoWorld& eegeoWorld)
```

Comment:
Seems very VR specific, can be moved elsewhere? Maybe even into CardboardExample?